packages/dappmanager/src/daemons/avahi/index.ts

```diff
@@ -1,80 +1,125 @@
+      // where -R argument is neccesary only if you want to publish more domains pointing to the same IP
+      // It runs in the foreground until it is sent SIGINT (ctrl+c) and during that time all machines
+      // on broadcast domain should be able to resolve that domain to the IP
+      // See https://github.com/dappnode/DNP_DAPPMANAGER/issues/669
+
+      // > Not necessary to manually SIGINT avahi-publish. DAPPMANAGER does not gracefully shutdown now
+      //   so Docker will just kill all processes eventually
+
+      // Broken pipe. If you use pipes or FIFOs, you have to design your application so that one process
+      // opens the pipe for reading before another starts writing. If the reading process never starts,
+      // or terminates unexpectedly, writing to the pipe or FIFO raises a SIGPIPE signal. If SIGPIPE is blocked,
+      // handled or ignored, the offending call fails with EPIPE instead.
+      // To dettach the process from the parent: https://stackoverflow.com/questions/25323703/nodejs-execute-command-in-background-and-forget
+
+      await retry(
+        bail =>
+          new Promise<void>((resolve, reject) => {
+            // Using child_process.spawn since it's better for long running processes with unbounded outputs
+            // - shell(): Is meant for short processes and features a timeout and buffer limit that we don't want here
+            // - child_process.exec: Buffers output, which can cause issues since we want avahi-publish to run for potentially weeks
+            const avahi = spawn(
+              "avahi-publish",
+              ["-a", "-R", params.AVAHI_LOCAL_DOMAIN, internalIp],
+              // Pass through the corresponding stdio stream to/from the parent process.
+              // In the first three positions, this is equivalent to process.stdin, process.stdout, and process.stderr, respectively
+              { stdio: ["inherit", "inherit", "inherit"] }
+            );
+
+            avahi.on("close", code => {
+              logs.info(`avahi-publish exited with code ${code}`);
+              if (code === 0) resolve();
+              else reject(Error(`Exit code ${code}`));
+            });
+
+            avahi.on("error", err => {
+              logs.error("avahi-publish error", err);
+              reject(err);
+            });
+
+            controller.signal.addEventListener("abort", () => {
+              bail(Error("Aborted")); // Prevent more retries
+              avahi.kill("SIGINT"); // Gracefully kill spawned process
+            });
+          }),
+        {
+          retries: 10,
+          maxRetryTime: Infinity,
+          maxTimeout: Infinity,
+          onRetry: e => logs.error(`Error on avahi-publish`, e)
+        }
+      );
+    } catch (e) {
+      logs.error("Too many errors on avahi-publish - stopping daemon", e);
+    } finally {
+      this.status.controller.abort();
+      this.status = { type: AvahiStatusType.crashed };
```

Comment:
This is not correct, the finally block will execute if the avahi spawned process exits with code 0 on shutdown